polar2grid/utils/legacy_compat.py

```diff
@@ -65,6 +63,13 @@ def convert_p2g_pattern_to_satpy(pattern):
         "begin_time": "start_time",
         "product_name": "p2g_name",
     }
+    # If there is no other formatting before replacing begin_time,
+    # add the old default formatting to the pattern and changed begin_time to
+    # start_time
+    abs_search = re.compile(r"{begin_time}").search
+    if bool(abs_search(pattern)):
```

Comment:
Are regular expressions needed? Or even this if statement? Would only doing:

```
pattern = pattern.replace("{begin_time}", "{start_time:%Y%m%d_%H%M}")
```

work?